plugin/src/main/java/org/gradle/testretry/internal/framework/TestNgTestFrameworkStrategy.java

```diff
@@ -59,7 +64,19 @@ public TestFramework createRetrying(JvmTestExecutionSpec spec, Test testTask, Se
                 }
             });
 
-        return new TestNGTestFramework(testTask, retriedTestFilter, instantiator, classLoaderCache);
+        VersionNumber gradleVersion = VersionNumber.parse(testTask.getProject().getGradle().getGradleVersion());
+        if (gradleVersion.getMajor() >= 6 && gradleVersion.getMinor()>= 6) {
+            final ObjectFactory objectFactory = ((ProjectInternal) testTask.getProject()).getServices().get(ObjectFactory.class);
+            return new TestNGTestFramework(testTask, objectFactory.fileCollection(), retriedTestFilter, objectFactory);
+        } else {
+            try {
+                Class<?> testNGTestFramework = Class.forName("org.gradle.api.internal.tasks.testing.testng.TestNGTestFramework");
```

Comment:
```suggestion
                Class<?> testNGTestFramework = TestNGTestFramework.class;
```

Reply:
This would assume that the package will not change in later versions - which I think is save.